.devcontainer/devcontainer.json

```diff
@@ -3,20 +3,24 @@
 {
     "name": "PowerShell",
     "dockerFile": "Dockerfile",
-        
     // Uncomment the next line if you want to publish any ports.
     // "appPort": [],
-
     // Uncomment the next line if you want to add in default container specific settings.json values
     // "settings":  { "workbench.colorTheme": "Quiet Light" },
-
     // Uncomment the next line to run commands after the container is created. This gets run in bash which is why we call `pwsh`.
     // "postCreateCommand": "pwsh -c '$PSVersionTable'",
-
     // Uncomment the next line to use a non-root user. See https://aka.ms/vscode-remote/containers/non-root-user.
     // "runArgs": [ "-u", "1000" ],
-
+    "runArgs": [
+        "-v",
+        "/var/run/docker.sock:/var/run/docker.sock"
```

Comment:
What does this do?

Reply:
We're using a technology called Docker-in-Docker.  It has many use cases like CI pipelines where you have an environment container and you're building a container for deployment in it.  

We are using a derivative of a CI pipeline for our dev environment.  Since we're distributing a Docker container with all our dependencies in it (via our VSCode Dev Container), we're running all our code during dev in a container.  

We now need to build docker containers to push into k8s.  Building a container involves running a trimmed down instance of that container temporarily. We essentially need to run docker _inside_ docker.  This is complicated.

Instead of putting a kernel in a kernel in a kernel in a hypervisor, this PR installs just the cli tools docker uses and not the fun emulation/virtualization stuff.  This specific block of code then maps the internal docker port that the cli talks to to the external port your host machine runs the virtualization daemon on.  This allows a docker command run from inside the container to build and access images and containers on the host.

Reply:
All in all, this dev container stuff is a super cool technology.  It makes it really easy for anyone to clone this repo and instantly be able to run all of our code and then modify it if it doesn't suit their exact needs.  IMO that's an ideal way to distribute not only to each other, but our target segment.